cv/lane_detection/src/lane_detection_model_inference.py

```diff
@@ -1,136 +1,149 @@
+    med4 = np.median(gray_im[height // 2 :, : width // 2])
 
-    l1 = int(max(0,(1-0.205)*med1))
-    u1 = int(min(255,(1+0.205)*med1))
-    e1 = cv2.Canny(gray_im[:height//2,:width//2],l1,u1)
+    l1 = int(max(0, (1 - 0.205) * med1))
+    u1 = int(min(255, (1 + 0.205) * med1))
+    e1 = cv2.Canny(gray_im[: height // 2, : width // 2], l1, u1)
 
-    l2 = int(max(0,(1-0.205)*med2))
-    u2 = int(min(255,(1+0.205)*med2))
-    e2 = cv2.Canny(gray_im[:height//2,width//2:],l2,u2)
+    l2 = int(max(0, (1 - 0.205) * med2))
+    u2 = int(min(255, (1 + 0.205) * med2))
+    e2 = cv2.Canny(gray_im[: height // 2, width // 2 :], l2, u2)
 
-    l3 = int(max(0,(1-0.205)*med3))
-    u3 = int(min(255,(1+0.205)*med3))
-    e3 = cv2.Canny(gray_im[height//2:,width//2:],l3,u3)
+    l3 = int(max(0, (1 - 0.205) * med3))
+    u3 = int(min(255, (1 + 0.205) * med3))
+    e3 = cv2.Canny(gray_im[height // 2 :, width // 2 :], l3, u3)
 
-    l4 = int(max(0,(1-0.205)*med4))
-    u4 = int(min(255,(1+0.205)*med4))
-    e4 = cv2.Canny(gray_im[height//2:,:width//2],l4,u4)
+    l4 = int(max(0, (1 - 0.205) * med4))
+    u4 = int(min(255, (1 + 0.205) * med4))
+    e4 = cv2.Canny(gray_im[height // 2 :, : width // 2], l4, u4)
 
     # Stitch the edges together
-    edges_mask[:height//2,:width//2] = e1
-    edges_mask[:height//2,width//2:] = e2
-    edges_mask[height//2:,width//2:] = e3
-    edges_mask[height//2:,:width//2] = e4
+    edges_mask[: height // 2, : width // 2] = e1
+    edges_mask[: height // 2, width // 2 :] = e2
+    edges_mask[height // 2 :, width // 2 :] = e3
+    edges_mask[height // 2 :, : width // 2] = e4
 
     edges_mask_inv = cv2.bitwise_not(edges_mask)
 
     return edges_mask, edges_mask_inv
 
 
-def get_input(frame):
-    frame = cv2.resize(frame,(1280,720),interpolation=cv2.INTER_AREA)
+def get_input(frame: np.ndarray) -> np.ndarray:
     frame_copy = np.copy(frame)
-
-    test_edges,test_edges_inv = find_edge_channel(frame_copy)
-    frame_copy = np.append(frame_copy,test_edges.reshape(test_edges.shape[0],test_edges.shape[1],1),axis=2)
-    frame_copy = np.append(frame_copy,test_edges_inv.reshape(test_edges_inv.shape[0],test_edges_inv.shape[1],1),axis=2)
-    frame_copy = cv2.resize(frame_copy,(330,180))
-
-    input = (frame_copy/255.).transpose(2,0,1).reshape(1,5,180,330)
+    frame_copy = cv2.resize(frame_copy, (WIDTH, HEIGHT), interpolation=cv2.INTER_AREA)
+
+    test_edges, test_edges_inv = find_edge_channel(frame_copy)
+    frame_copy = np.block(
+        [
+            frame_copy,  # shape: (330, 180, 3)
+            test_edges.reshape(WIDTH, HEIGHT, 1),  # shape: (330, 180, 1)
+            test_edges_inv.reshape(WIDTH, HEIGHT, 1),  # shape: (330, 180, 1)
+        ]
+    )  # (330, 180, 5)
+
+    # Is this the correct permutation? Maybe try it out
+    # Shapes go: (330, 180, 5) ----> (5, 330, 180) --ERROR??--> (1, 5, 180, 330)
+    input = (frame_copy / 255.0).transpose(2, 0, 1).reshape(1, 5, HEIGHT, WIDTH)
```

Comment:
@Jason-Y000 please double-check the logic and see if it's a bug or I'm just tripping. I'm really confused rn ngl and my brain is about to turn off from utter exhaustion 😫 lol

I'm just worried that it is changing the shape in a bad way (transposing the 330 with the 180)